src/de/jost_net/JVerein/server/DDLTool/Updates/Update0447.java

```diff
@@ -0,0 +1,61 @@
+/**********************************************************************
+ * This program is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation, either version 3 of the 
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without 
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See 
+ * the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program. If not, 
+ * see <http://www.gnu.org/licenses/>.
+ * 
+ **********************************************************************/
+package de.jost_net.JVerein.server.DDLTool.Updates;
+
+import de.jost_net.JVerein.server.DDLTool.AbstractDDLUpdate;
+import de.jost_net.JVerein.server.DDLTool.Column;
+import de.willuhn.util.ApplicationException;
+import de.willuhn.util.ProgressMonitor;
+
+import java.sql.Connection;
+
+public class Update0447 extends AbstractDDLUpdate
+{
+  public Update0447(String driver, ProgressMonitor monitor, Connection conn)
+  {
+    super(driver, monitor, conn);
+  }
+
+  @Override
+  public void run() throws ApplicationException
+  {
+    execute(alterColumn("mail",
+        new Column("versand", COLTYPE.TIMESTAMP, 0, "NULL", false, false)));
+
+    execute(alterColumn("mailempfaenger",
+        new Column("versand", COLTYPE.TIMESTAMP, 0, "NULL", false, false)));
+
+    execute(dropForeignKey("fkAnfangsbestand1", "anfangsbestand"));
+
+    execute(createForeignKey("fkAnfangsbestand1", "anfangsbestand", "konto",
+        "konto", "id", "CASCADE", "RESTRICT"));
+
+    execute(dropForeignKey("fkLehrgang2", "lehrgang"));
+
+    execute(createForeignKey("fkLehrgang2", "lehrgang", "lehrgangsart",
+        "lehrgangsart", "id", "RESTRICT", "RESTRICT"));
+
+    execute(createForeignKey("fkBuchung9", "buchung", "splitid", "buchung",
+        "id", "RESTRICT", "RESTRICT"));
+
+    execute(createForeignKey("fkMitgliednextbgruppe2", "mitgliednextbgruppe",
+        "mitglied", "mitglied", "id", "CASCADE", "RESTRICT"));
+
+    execute(alterColumn("buchungsart", new Column(
+        "steuer_buchungsart", COLTYPE.BIGINT, 10, "NULL", false, false)));
+
+    execute(createForeignKey("fkBuchungsart3", "buchungsart",
+        "steuer_buchungsart", "buchungsart", "id", "RESTRICT", "RESTRICT"));
```

Comment:
Das hier ist etwas seltsam. Das Attribut steuer_buchungsart ist aktuell als VARCHAR(10) definiert. Als Referenz hätte es ein BIGINT sein sollen.
Nach der Migration ist es ein STEUER_BUCHUNGSART BIGINT DEFAULT NULL. D.h. die Migration hat den Typ geändert.
Der Code betrachtet es immer noch als VARCHAR. Es gibt eine Exception wenn man jetzt eine Buchungsart mit gesetzter Steuerart öffnet.

[Mon Oct 21 09:58:08 CEST 2024][ERROR][main][de.willuhn.jameica.gui.GUI$5.run] error while loading view de.jost_net.JVerein.gui.view.BuchungsartView
java.lang.ClassCastException: class java.lang.Long cannot be cast to class java.lang.String (java.lang.Long and java.lang.String are in module java.base of loader 'bootstrap')
	at de.jost_net.JVerein.server.BuchungsartImpl.getSteuerBuchungsart(BuchungsartImpl.java:210)
	at de.jost_net.JVerein.gui.control.BuchungsartControl.getSteuerBuchungsart(BuchungsartControl.java:327)
	at de.jost_net.JVerein.gui.view.BuchungsartView.bind(BuchungsartView.java:45)

Reply:
stimmt, habe ich behoben